airsonic-main/src/main/java/org/airsonic/player/dao/LegacyHsqlDaoHelper.java

```diff
@@ -18,6 +35,9 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(LegacyHsqlDaoHelper.class);
 
+    @Autowired
```

Comment:
This uses field injection.

Maybe consider putting this on the setter so a method injector is used?
Or just remove the getter/setter altogether?

Reply:
Airsonic's getter / setter injections are mostly bad habits and can be removed.
On the other hand partially, there are places where getters / setters are used for delayed injection.
(Without using @Lazy)

Because there is no advantage, I agree with the opinion of eliminating all getter / setter injections.

Reply:
Don't think getter/setters are bad habits, in fact, I think they're actually recommended over field injection (which is considered harmful). Regardless, discussions over code purity aside, we should pick one and either put the Autowired over the setter if we're putting a setter out there, or remove the setter if we're doing field injection.

Reply:
The reason why the Spring team deprecates field injection is because it recommends constructor injection.
(setter / setter allows the most ambiguous description for circular references.)
I think it's okay to turn off either one here.

Reply:
Hmm, actually, Spring promotes both constructor and setter injection based on mandatory or optional nature of dependencies respectively. It's just field injection it doesn't recommend (though it's not deprecated)

Field injections are reflection-based and have issues with immutable fields along with some more semantic problems.

In any case, it seems we're in agreement here, to remove one of the two.

Reply:
Which one do you guys think is better? I actually don't see a lot of issues with using field injection like this in Airsonic, it usually works well.

(Also, we should probably do this in another PR if we want to change the standard everywhere in Airsonic.)

Reply:
Spring would recommend setter/constructor injection
I personally think it's better to keep it consistent, and change it with everything else if we choose to, so field injection.

Reply:
It may be better to use a setter here (often used now).

If the standard is changed in the future, the setter can be stopped and rewritten as a constructor.
In Airsonic, most of the injection fields can be encapsulated, with some exceptions.
(Except when adopting special design patterns, there is no necessity to publish as public. )

Reply:
I actually removed `settingsService` entirely, it's not needed anymore since the logic has been moved to `Application` startup.